shell/platform/common/cpp/client_wrapper/plugin_registrar_unittests.cc

```diff
@@ -179,4 +236,37 @@ TEST(PluginRegistrarTest, ManagerRemovesOnDestruction) {
             nullptr);
 }
 
+// Tests texture register that calls through to the C API.
+TEST(MethodCallTest, RegisterTexture) {
+  testing::ScopedStubFlutterApi scoped_api_stub(std::make_unique<TestApi>());
+  auto test_api = static_cast<TestApi*>(scoped_api_stub.stub());
+
+  auto dummy_registrar_handle =
+      reinterpret_cast<FlutterDesktopPluginRegistrarRef>(1);
+  PluginRegistrar registrar(dummy_registrar_handle);
+  TextureRegistrar* textures = registrar.textures();
+
+  EXPECT_EQ(test_api->last_texture_id(), -1);
+  auto texture = test_api->GetFakeTexture(0);
+  EXPECT_EQ(texture, nullptr);
+
+  int64_t texture_id = textures->RegisterTexture(reinterpret_cast<Texture*>(2));
+  EXPECT_EQ(test_api->last_texture_id(), texture_id);
+  EXPECT_EQ(test_api->textures_size(), static_cast<size_t>(1));
+
+  texture = test_api->GetFakeTexture(texture_id);
+  EXPECT_EQ(texture->texture_id, texture_id);
+  EXPECT_EQ(texture->user_data, reinterpret_cast<Texture*>(2));
+
+  textures->MarkTextureFrameAvailable(texture_id);
+  textures->MarkTextureFrameAvailable(texture_id);
+  textures->MarkTextureFrameAvailable(texture_id);
+  EXPECT_EQ(texture->mark_count, 3);
+
+  textures->UnregisterTexture(texture_id);
+  texture = test_api->GetFakeTexture(texture_id);
+  EXPECT_EQ(texture, nullptr);
+  EXPECT_EQ(test_api->textures_size(), static_cast<size_t>(0));
+}
```

Comment:
This is a test of the texture registrar; it should be in a texture_registrar_unittests.cc file. The test here should just be that the getter on PluginRegistrar works.

Reply:
Also, there should be more than a single test. For instance, anything with error return values should have coverage where feasible (e.g., if unregister is going to return a bool, then there should be a test that unregistering a non-existent texture returns false as expected).